app/models/manageiq/providers/base_manager/metrics_capture.rb

```diff
@@ -175,18 +175,41 @@ def create_rollup_task_for_cluster(ems_cluster, hosts)
     end
   end
 
+  # number of records to send at a time for a single capture message
+  # @return [Numeric] 0 for no batching.  (defaults to 0)
+  def batch_size
+    @batch_size ||= ::Settings.dig(:ems, "ems_#{ems.provider_name.underscore}", :capture_batch_size).to_i
+  end
+
   def perf_capture_queue_targets(targets, interval, start_time:, end_time:, parent: nil)
-    task = create_rollup_task_for_cluster(parent, targets) if parent
+    if batch_size > 1
+      if parent # send all of the hosts in a cluster in one message
+        perf_capture_queue_target(targets, interval, :start_time => start_time, :end_time => end_time, :rollup => true)
+      else # batch sends
+        targets.each_slice(batch_size) do |targets_batch|
+          perf_capture_queue_target(targets_batch, interval, :start_time => start_time, :end_time => end_time)
+        end
+      end
```

Comment:
Possible refactoring to DRY up

I also added a comment about _why_ we send all of the hosts in a cluster in one message.  The previous comment doesn't describe that and I think it's important to add that even if we don't go with my refactoring below.

```suggestion
      # send all of the hosts in a cluster in one message and ignore batching
      #   hosts will rollup into the cluster, so the entire batch must be together otherwise
      #   we will require a create_rollup_task_for_cluster
      batch_size = targets.size if parent 

      targets.each_slice(batch_size) do |targets_batch|
        perf_capture_queue_target(targets_batch, interval, :start_time => start_time, :end_time => end_time)
      end
```

Reply:
I like the comment. Put something into the commit message but this needs to be in the code.

I did have the `batch_size = targets.size` in the previous rev, but it will create a bunch of extra arrays -  probably negligible. 

I'll make the change but since there are 2 variables in play, I thought it was easier to read with 2 calls to perf_capture_target (with the extra `rollup` parameter)